pkg/auth/handler/forgotpwd/reset_form.go

```diff
@@ -257,15 +257,12 @@ func (h ForgotPasswordResetFormHandler) HandleResetSuccess(rw http.ResponseWrite
 }
 
 func (h ForgotPasswordResetFormHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	_, err := handler.Transactional(h.TxContext, func() (_ interface{}, err error) {
-		err = h.Handle(w, r)
-		if err == nil {
-			err = h.HookProvider.WillCommitTx()
-		}
-		return
+	h.TxContext.UseHook(h.HookProvider)
+	_, err := handler.Transactional(h.TxContext, func() (interface{}, error) {
```

Comment:
I have an idea to have a further cleanup here.

- Move `db.WithTx` to `TxContext.Do`
- Move `handler.Transactional` to `TxContext.DoInterface`

What do you think?

Reply:
I think there's no need to move these. These functions are implemented using public API of `TxContext` (like extension method in C#), so their implementation should not be moved into specific implementation of `TxContext`.